File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

def DFS(v, graph, ch):
    global cnt
    cnt += 1
    ch[v] = 1
    for nv in graph[v]:
        if ch[nv] == 0:
            DFS(nv, graph, ch)

n = int(sys.stdin.readline().strip())
m = int(sys.stdin.readline().strip())

# 인접 리스트 생성
graph = [[] for _ in range(n+1)]
# 카운팅한 컴퓨터 확인하기 위한 변수
ch = [0] * (n+1)

cnt = 0

# 양방향이므로 둘 다 저장
for _ in range(m):
    i, j = map(int, sys.stdin.readline().strip().split())
    graph[i].append(j)
    graph[j].append(i)

# 1번 컴퓨터에 직접적으로 연결된 컴퓨터들부터
DFS(1, graph, ch)

# 1을 제외하기 위해 -1
print(cnt-1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

public class Main {
    static int n, m, cnt = 0;
    static List<Integer>[] graph;
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        m = Integer.parseInt(br.readLine());

        graph = new List[n+1];
        // 인접 리스트 초기화
        for (int i = 0; i <= n; i++)    graph[i] = new ArrayList();


        // 양방향 연결 저장
        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a].add(b);
            graph[b].add(a);
        }

        visited = new boolean[n + 1]; // 방문 체크 배열

        // DFS 시작
        DFS(1);

        // 1번 컴퓨터를 제외한 감염된 컴퓨터 수 출력
        System.out.println(cnt - 1);

    }

    public static void DFS(int v) {
        visited[v] = true;
        cnt++;

        for (int nv : graph[v]) {
            if (!visited[nv]) {
                DFS(nv);
            }
        }
    }
}